docs/commands.md

```diff
@@ -0,0 +1,153 @@
+
+The namespace defaults to the value in `.faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `.faas.yaml`. The user may specify a repository on the command line.
+
+Note that the behavior of `update` is different than that of `deploy` and `run`.  When `update` is run, a new container image is always built. However, for `deploy` and `run`, the user is required to run `faas build` first. The `update` command also differs from `deploy` in that it allows the user to specify a repository on the command line (but still not an image name). Consider normalizing all of this so that all of these commands behave similarly.
+
+Similar `kn` command: `kn service update NAME [flags]`. As with `deploy`, the `update` command provides a level of simplicity for a new user that restricts flexibility while improving the ease of use.
+
+```console
+faas update [-r <repository> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas update [-r <repository> -p <path>]
+```
+
+## `describe`
+
+Prints the name, route and any event subscriptions for a deployed Function. The user may also specify the name of the function to describe, and the namespace in which to search for the function. If provided, this will be updated in the project's `.faas.yaml` configuration file. The user may also supply a path to the Function project.
+
+Similar `kn` command: `kn service describe NAME [flags]`. This flag provides a lot of nice information not available in `faas describe`, such as revisions, age, annotations and labels. Consider a way to combine the two outputs, or rename this command to make it distinct from `kn` - e.g. `faas info`.
+
+```console
+faas describe [-f <format> -n <namespace> -p <path>]
+```
+
+When run as a `kn` plugin.
+
+```console
+kn faas describe [-f <format> -n <namespace> -p <path>]
+```
+
+## `list`
+
+Lists all deployed functions. The namespace defaults to the value in `.faas.yaml` or the namespace currently active in the user's Kubernetes configuration. The namespace may be specified on the command line, and if so this will overwrite the value in `.faas.yaml`.
+
+Similar `kn` command: `kn service list [name] [flags]`. This command lists all deployed Knative `Services`. As with other `kn` commands that have similar functionality, there is more information and flexibilty in the `kn` command. However, `kn` will return _all_ `Services`, while `faas list` will only display the boson Functions that have been deployed. Consider improving the output of the `faas list` command so that it is at least as informative as `kn service list`.
```

Comment:
> Consider improving the output of the `faas list` command so that it is at least as informative as `kn service list`.

I would recommend we leave this as a very plain, simple list of names, such that it can be easily integrated with other systems via a shell script.  The details about the services would then be available through the same mechanism using `faas status <name>`
```
$ faas list
www.boson-project.org
admin.boson-project.org
$ faas list | xargs faas status
... details of each
```

In particular:  `faas list | xargs faas status`

Reply:
Oh, I see that the `faas status` no longer accepts the name as input.  We might want to consider adding that back in.  In particular I am thinking this is useful for someone wanting to get details about a function for which they do not currently have the code locally, but do have access to the cluster.

Reply:
Upon further thought, I like the idea of adding the extra data, so long as that first token is the name which is the input to `status` such that it's still easily scriptable.